engine/dlib/src/dlib/connection_pool.cpp

```diff
@@ -375,20 +378,24 @@ namespace dmConnectionPool
             if (!FindSlot(pool, &index, &c)) {
                 return RESULT_OUT_OF_RESOURCES;
             }
-            c->m_State = STATE_INUSE;
+            c->m_State = STATE_INUSE; // The connection might fail, but we need to reserve this in the meantime
 
         } else {
             *sock_res = dmSocket::RESULT_HOST_NOT_FOUND; // Only error that dmSocket::getHostByName returns
             return RESULT_SOCKET_ERROR;
         }
 
-        Result r = Connect(pool, host, address, port, ssl, timeout, c, sock_res);
+        dmSocket::Socket socket = dmSocket::INVALID_SOCKET_HANDLE;
+        dmSSLSocket::Socket sslsocket = dmSSLSocket::INVALID_SOCKET_HANDLE;
+        Result r = Connect(pool, host, address, port, ssl, timeout, &socket, &sslsocket, sock_res);
```

Comment:
This part is unprotected (to minimize the time the lock is being held).
So, we're not allowed to read/write from the struct or TSan will complain.
In practice I don't think this was an actual problem, but more of a correctness issue.